data_collection/gazette/spiders/sp_americana.py

```diff
@@ -0,0 +1,68 @@
+import dateparser
+from scrapy import Spider
+import datetime
+from dateutil.relativedelta import relativedelta
+from gazette.items import Gazette
+import re
+
+
+class SpAmericanaSpider(Spider):
+    TERRITORY_ID = "3501608"
+    name = "sp_americana"
+    allowed_domains = ["diariooficial.americana.sp.gov.br"]
+    start_date = datetime.date(2018, 9, 1)  # First gazette available
+    start_urls = ["https://diariooficial.americana.sp.gov.br/diario-oficial-edicaoAnterior.php"]
+    extra_editions = "https://diariooficial.americana.sp.gov.br/diario-oficial-edicaoExtra.php"
+
+    locations = {
+        "gazette": "//*[@class='day table-light']/strong/a",
+        "extra_gazette_url": "//*[@class='list-group-item']/a/@href",
+        "extra_gazette_date": "//*[@class='list-group-item']/text()"}
+
+    def parse(self, response):
+
+        param_url = f"?mes={self.start_date.month}&ano={self.start_date.year}"
+        base_url = response.url
+        url = base_url + param_url
+        date = self.start_date
+
+        while date.year < datetime.date.today().year or date.month < datetime.date.today().month:
```

Comment:
**suggestion (non-blocking)** Instead of calculating `datetime.date.today()` every loop iteration, you could set a new auxiliary variable and calculate it only once:
```
today = datetime.date.today()
while <something>:
    <do_something>

Reply:
really much better and clean, adjusted